syntax/printer.go

```diff
@@ -39,24 +50,37 @@ func SpaceRedirects(p *Printer) { p.spaceRedirects = true }
 // Note that this feature is best-effort and will only keep the
 // alignment stable, so it may need some human help the first time it is
 // run.
-func KeepPadding(p *Printer) {
-	p.keepPadding = true
-	p.cols.Writer = p.bufWriter.(*bufio.Writer)
-	p.bufWriter = &p.cols
+func KeepPadding(enabled bool) PrinterOption {
+	return func(p *Printer) {
+		if enabled {
+			p.keepPadding = true
+			p.bufWriter = &p.cols
+		} else {
+			p.keepPadding = false
+			p.bufWriter = p.cols.Writer
+		}
+	}
 }
 
 // Minify will print programs in a way to save the most bytes possible.
 // For example, indentation and comments are skipped, and extra
 // whitespace is avoided when possible.
-func Minify(p *Printer) { p.minify = true }
+func Minify(enabled bool) PrinterOption {
+	return func(p *Printer) { p.minify = enabled }
+}
 
 // NewPrinter allocates a new Printer and applies any number of options.
-func NewPrinter(options ...func(*Printer)) *Printer {
+func NewPrinter(opts ...PrinterOption) *Printer {
+	bw := bufio.NewWriter(nil)
+
 	p := &Printer{
-		bufWriter: bufio.NewWriter(nil),
+		bufWriter: bw,
 		tabWriter: new(tabwriter.Writer),
+		cols: colCounter{
+			Writer: bw,
+		},
```

Comment:
why this change? it's not mentioned at all in the commit message, and I don't see why it's necessary

Reply:
This is to make KeepPadding option togglable, so that user can call it with false or true in any order any number of times.

Reply:
Previously setting KeepPadding multiple times would result in a panic.

Reply:
If that's the case, please file a separate issue for that, and let's fix that in a separate PR with a test.

Reply:
It's also possible to remove writer swapping logic and always write to colCounter irregardless of KeepPadding setting. This would simplify code slightly, but also slightly regress performance.

Reply:
I think we should discuss that in a separate issue+PR; this thread and patch are big enough as it is.

Reply:
I've changed this back and added TODO comment.